### PR TITLE
update imageconfig on service creation/update

### PR DESCRIPTION
**Pull request by bparees**
this PR extracts the internal registry hostname from the observed service object and reflects it into the global Image configuration object for the cluster.

### CODE REVIEW

File: pkg/operator/handler.go [hunk 1]
```diff
@@ -305,6 +305,14 @@ func (h *Handler) Handle(ctx context.Context, event sdk.Event) error {
 		if err != nil {
 			return err
 		}
+		svc := event.Object.(*coreapi.Service)
+		if cr != nil {
+			svcHostname := fmt.Sprintf("%s.%s.cluster.svc:%d", svc.Name, svc.Namespace, svc.Spec.Ports[0].Port)
```
Comment (dmage) [inline pkg/operator/handler.go]: `.cluster.svc`? Is it something new in 4.0?

File: deploy/00-crd.yaml [hunk 2]
```diff
@@ -12,3 +12,17 @@ spec:
...
+  name: images.config.openshift.io
+spec:
+  group: config.openshift.io
+  scope: Cluster
+  version: v1
+  names:
+    kind: Image
+    listKind: ImageList
+    plural: images
+    singular: image
```
[the remark this answers is not in the document]
Comment (legionus) [reply deploy/00-crd.yaml]: @bparees Why is it called a not `image-registry` ?

File: pkg/operator/handler.go (hunk 1 above)
Comment (legionus) [reply pkg/operator/handler.go]: @bparees you mean `my-svc.my-namespace.svc.cluster.local` ?

https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#services

File: deploy/00-crd.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (bparees) [reply deploy/00-crd.yaml]: I considered the same, but this is why we have api groups.  We need to accept there will be duplicate resource types across different api groups.  The name was already agreed on here:  https://github.com/openshift/api/pull/111

File: pkg/operator/handler.go (hunk 1 above)
Comment (bparees) [reply pkg/operator/handler.go]: sorry yes i got it backwards.

Comment (bparees) [reply pkg/operator/handler.go]: pushed fix.

File: pkg/operator/handler.go [hunk 3]
```diff
@@ -52,6 +52,7 @@ func NewHandler(namespace string) (sdk.Handler, error) {
 	p.Healthz.TimeoutSeconds = 5
 
 	p.Service.Name = "image-registry"
+	p.ImageConfig.Name = "cluster"
```
Comment (bparees) [inline pkg/operator/handler.go]: @deads2k @juanvallejo  relevant to your "observe the internal registry hostname" PR.  

this PR adds the logic to the image-registry operator to publish the config object, named "cluster" (per @deads2k's "canonical" comment on the api type)

**Comment by gabemontero** [comment]
@bparees - is the change from this merged PR sufficient for the sample operator to determine if the registry is up before creating/updating imagestreams?

Seems like sample operator could create a watch on the new ImageConfig object now being generated, and look for the internal registry hostname to be set, no?

File: pkg/operator/handler.go (hunk 3 above)
Comment (juanvallejo) [reply pkg/operator/handler.go]: @bparees thanks, I'll go ahead and update https://github.com/openshift/cluster-openshift-apiserver-operator/pull/41 accordingly

**Pull request merged**